warehouse/ingest-core/src/main/java/datawave/ingest/table/aggregator/GlobalIndexUidAggregator.java

```diff
@@ -150,18 +144,21 @@ public Value reduce(Key key, Iterator<Value> iter) {
                 long delta = v.getCOUNT();
                 
                 count += delta;
-                /**
+                /*
                  * Fail fast approach.
                  */
                 if (v.getIGNORE()) {
                     seenIgnore = true;
-                    if (log.isDebugEnabled())
-                        log.debug("SeenIgnore is true. Skipping collections");
+                    log.debug("SeenIgnore is true. Skipping collections");
                 }
                 
-                // if delta > 0, we are collecting the uid list
-                // in the protobuf into our object's uid list.
-                if (delta > 0) {
+                // if delta >= 0, we are collecting the uid list in the protobuf into our object's uid list.
+                //
+                // Note that it is possible to have a delta of 0 if we were to have gone through
+                // a partial major compaction and combined a protocol buffer having a positive
+                // count of 1 and a negative count of 1 (leaving one UID in the UID list and one
+                // UID in the REMOVEDUID list.
+                if (delta >= 0) {
```

Comment:
Nice catch. I'm wondering if this could even happen with numbers less than 0 - like if one RFile contains an insert of uid1 and another RFile has a removeUidList with uid2 and uid3 in it.

Reply:
@jschmidt10 Good point. Maybe the same action needs to be taken regardless of the count in the incoming protocol buffer -- add the UIDs in the add list, and remove the ones in the remove list. I also have some stuff in here that changes behavior w.r.t timestamps (I'm assuming timestamp matters but I know by default datawave truncates timestamp to the day boundary so I'll need to guard that behavior change with an option). I'll take a look at all that at some point and push some updates.

Reply:
Sounds good. Thanks!